src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -118,7 +118,8 @@
      (format "sweep of stale reports (threshold: %s)"
              (format-period report-ttl))
      (jdbc/with-transacted-connection db
-       (scf-store/delete-reports-older-than! (ago report-ttl))))
+       (scf-store/delete-reports-older-than! (ago report-ttl))
+       (scf-store/delete-old-unassociated-catalogs! (ago report-ttl))))
```

Comment:
Right, we'll also need to add an endpoint for sync to be able to pull down all the historical catalogs.